bitnami/common/templates/_storage.tpl

```diff
@@ -9,13 +9,12 @@ Return  the proper Storage Class
 {{ include "common.storage.class" ( dict "persistence" .Values.path.to.the.persistence "global" $) }}
 */}}
 {{- define "common.storage.class" -}}
-
-{{- $storageClass := .persistence.storageClass -}}
-{{- if .global -}}
-    {{- if .global.storageClass -}}
-        {{- $storageClass = .global.storageClass -}}
-    {{- end -}}
-{{- end -}}
+{{/*
+Uses https://stackoverflow.com/a/68807258/1306020 to avoid lots of ifs.
+This allows all storage classes to be overridden by .global.storageClass, and to
+fallback to .global.defaultStorageClass if no specific class was set.
+*/}}
+{{- $storageClass := (.global).storageClass | default .persistence.storageClass | default (.global).defaultStorageClass | default "" -}}
```

Comment:
Please correct me if I'm wrong but this will fail if `global` object isn't defined since you're trying to access a "storageClass" property of a `null` object.
Try with this:
```suggestion
{{- $defaultStorageClass := ternary (default "" .global.storageClass) "" .global -}}
{{- $storageClass := default $defaultStorageClass .persistence.storageClass -}}
```

Reply:
Sorry, got pulled on to other projects so this dropped down the my todo list. My understanding from the linked stackoverflow answer (though I'm very new to how the helm/go templating language works) is you can do `(.maybe_null).also_maybe_null` and this won't error due to the extra parentheses? Happy to use your flow if you think it's better.

Reply:
You're definitely right! Indeed, I love this syntax, I sent a PR to adopt it broadly in the `common` library, see https://github.com/bitnami/charts/pull/25446